monix-execution/shared/src/test/scala/monix/execution/schedulers/UncaughtExceptionReporterSuite.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2014-2019 by The Monix Project Developers.
+ * See the project homepage at: https://monix.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package monix.execution.schedulers
+
+import scala.concurrent.{ExecutionContext, Promise}
+import scala.concurrent.duration._
+
+import minitest.TestSuite
+import monix.execution.{ExecutionModel, FutureUtils, Scheduler, UncaughtExceptionReporter}
+
+
+class UncaughtExceptionReporterBaseSuite extends TestSuite[Promise[Throwable]] {
+  object Dummy extends Throwable
+  private[this] val throwRunnable: Runnable = new Runnable {
+    def run(): Unit = throw Dummy
+  }
+
+  private[this] val immediateEC = new ExecutionContext {
```

Comment:
I think we could use `monix.execution.schedulers.TrampolineExecutionContext.immediate`